### PR TITLE
SHOTGUN-6 update tk-config-unity to latest tk-config-default2 (1.3.16)

### PR DESCRIPTION
**Pull request by vkovec**
This also fixes an issue where tk-config-unity did not work with the latest Shotgun (issue was versions in engine_locations.yml and frameworks.yml out of date).

### CODE REVIEW

File: info.yml [hunk 1]
```diff
@@ -16,4 +16,4 @@ description: "A config for Unity-based pipelines. Based on tk-config-default2."
 
 # Required minimum versions for this item to run
 requires_shotgun_version: "v7.2.0"
-requires_core_version: "v0.19.6"
+requires_core_version: "v0.19.5"
```
Comment (benoithudson) [inline info.yml:19]: Is this right, we're downgrading the required core version?

Comment (vkovec) [reply info.yml:19]: 0.19.5 is what is used in tk-config-default2. Also this is the minimum required version. Actual version used is v0.19.17 (in core/core_api.yml) if I understood correctly.

**Review by benoithudson — APPROVED** [review]

File: info.yml (hunk 1 above)
Comment (lassond) [reply info.yml:19]: Leave it like that (`v0.19.5`)

As you say the actual tk-core version is `v0.19.7`, and studios can decide to change that. But if they want to use `tk-config-unity` as their pipeline config they will probably fail if they use something lower than `v0.19.5`. So they will edit `info.yml` to reduce the number and I am pretty sure everything will be fine :-)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    # Makes sure requirements.txt is properly formatted
+    - id: requirements-txt-fixer
+    # Removes trailing whitespaces.
+    - id: trailing-whitespace
+  # Leave black at the bottom so all touchups are done before it is run.
+  - repo: https://github.com/ambv/black
+    rev: 20.8b1
+    hooks:
+    - id: black
+      language_version: python3
```
Comment (lassond) [inline .pre-commit-config.yaml:41]: I wonder what that does??

File: env/includes/engine_locations.yml [hunk 3]
```diff
@@ -19,23 +19,34 @@ engines.tk-3dsmaxplus.location:
   name: tk-3dsmaxplus
   version: v0.5.10
 
+engines.tk-3dsmax.location:
+  type: app_store
+  name: tk-3dsmax
+  version: v1.1.4
+
```
Comment (lassond) [inline env/includes/engine_locations.yml:26]: I am surprised that 3DSMax is getting include so late

Comment (lassond) [reply env/includes/engine_locations.yml:26]: ah, must be something to do with 3dsmaxplus vs 3dsmax...

**Review by lassond — APPROVED** [review]
Great work, approved!
I guess release tasks are next, ping me if you need help with that

I am glad someone else is doing it this time, it will improve our release procedures.

**Pull request merged**